mreg_cli/network.py

```diff
@@ -203,6 +185,168 @@ def info(args):
+        cli_warning(f"Unable to display network information about a {t} object")
+
+    used = get_network_used_count(ip_range)
+    unused = get_network_unused_count(ip_range)
+    ip_network = ipaddress.ip_network(ip_range)
+
+    # Pretty print all network info
+    print_network(network_info["network"], "Network:")
+    print_network(ip_network.netmask.exploded, "Netmask:")
+    print_network(network_info["description"], "Description:")
+    print_network(network_info["category"], "Category:")
+    print_network(network_info["location"], "Location:")
+    print_network(network_info["vlan"], "VLAN")
+    print_network(
+        network_info["dns_delegated"] if network_info["dns_delegated"] else False,
+        "DNS delegated:",
+    )
+    print_network(network_info["frozen"] if network_info["frozen"] else False, "Frozen")
+    print_network_reserved(network_info["network"], network_info["reserved"])
+    print_network_excluded_ranges(network_info["excluded_ranges"])
+    print_network(used, "Used addresses:")
+    print_network_unused(unused)
+    cli_info(f"printed network info for {ip_range}")
+
+
+########################################
+# Implementation of sub command 'find' #
+########################################
+
+
+def find(args: Namespace):
+    """List networks matching search criteria."""
+    args_dict = vars(args)
+
+    params = {}
+    param_names = [
+        "network",
+        "description",
+        "vlan",
+        "dns_delegated",
+        "category",
+        "location",
+        "frozen",
+        "reserved",
+    ]
+    for name in param_names:
+        value = args_dict.get(name)
+        if value is None:
+            continue
+        param, val = convert_wildcard_to_regex(name, value)
+        params[param] = val
+
+    if not params:
+        cli_warning("Need at least one search criteria")
+
+    path = f"/api/v1/networks/"
+    networks = get_list(path, params)
+    if not networks:
+        cli_warning("No networks matching the query were found.")
+
+    n_networks = len(networks)
+    for i, nwork in enumerate(networks):
+        if args.limit and i >= args.limit:
+            omitted = n_networks - i
+            s = "s" if omitted > 1 else ""
+            if not args.silent:
+                print(f"Reached limit ({args.limit}). Omitted {omitted} network{s}.")
+            break
+        if args.addr_only:
+            print(nwork["network"])
+        else:
+            print_network_info(nwork)
+            print()  # Blank line between networks
+
+    s = "s" if n_networks > 1 else ""
+    if not args.silent:
+        print(f"Found {n_networks} network{s} matching the search criteria.")
+
+
+network.add_command(
+    prog="find",
+    description="Search for networks based on a range of search parameters.",
+    short_desc="Search for networks.",
+    callback=find,
+    flags=[
+        Flag(
+            "-network",
+            description="Network address.",
```

Comment:
Looks better without the ending punctation marks in the `desciption` and `short_desc`.

Reply:
Resolved in e71f4cb.